### DISCUSSION TITLE
How to convert multiple photos at once?

### DISCUSSION THREAD

**Discussion by MrPrometheus** [ImageMagick/ImageMagick#4563] [Help] [answered]
Hello, I want to know if it is possible to somehow convert several photos at once.
For example, there are 3 photos and you need to convert these 3 photos for 1 console command

**Comment by snibgo** [comment]
Yes, it is possible.

If you show us the three commands you use, we can show you how to combine them.

**Reply by MrPrometheus** [reply]
I use in cmd: Magick *.jpg *.png
And I have several photos, I am doing this command for each photo separately. Is it fashionable to do everything with one command?
latest version program

**Comment by fmw42** [comment, marked as answer]
Is that 3 inputs and 3 outputs or 3 inputs and one output?  What command did you use? What is your ImageMagick version and platform?

**Reply by MrPrometheus** [reply]
I use in cmd: Magick *.jpg *.png 
And I have several photos, I am doing this command for each photo separately.  Is it fashionable to do everything with one command?
latest version program

**Reply by fmw42** [reply]
Unfortunately, ImageMagick Magick command does not work that way. It expects only one output name. To do what you want, use 

`magick mogrify -format png *.jpg`

mogrify will read each jpg and then convert to png, one image at a time. It is like looping over each jpg in your directory with a FOR loop and writing the corresponding output. But you do not have to write the FOR loop.

**Reply by MrPrometheus** [reply]
Thanks this works for me. But this command has been working for a long time, I don’t know why.
I wrote a program in C # that also takes each directory file and converts it to the desired format. It happens much faster. In my program I use the command - magick *.jpg *.png
